cmd/tea/gen_id.go

```diff
@@ -214,28 +210,74 @@ func (i IDs) generateUUID() ([]byte, error) {
 	return generatedOutput.Bytes(), nil
 }
 
-func (i IDs) generateHeader() []byte {
-	var d []byte
-	d = append(d, "package id\n\n"...)
-	d = append(d, "import (\n"...)
-	d = append(d, "\t\"fmt\"\n"...)
+func (i IDs) generateStringID() ([]byte, error) {
+	ids, ok := i["string"]
+	if !ok {
+		return nil, nil
+	}
+
+	generatedOutput := &bytes.Buffer{}
+	data := map[string][]string{
+		"ids": ids,
+	}
 
+	t, err := template.New("string").Parse(stringTemplate)
+	if err != nil {
+		return nil, err
+	}
+	if err = t.Execute(generatedOutput, data); err != nil {
+		return nil, err
+	}
+
+	return generatedOutput.Bytes(), nil
+}
+
+func (i IDs) generateHeader() []byte {
```

Comment:
Not sure if there were any plans with this function, i.e extending it with custom standard or external imports, but isn't this unnecessarily complicated? Can we just use `embed` and maintain the headers in a separate file?

Reply:
There is the complication that each ID type requires sometimes different and sometimes the same imports. And when you want only string ID, there are no imports, with uint64, there is no uuid import (a lot of different combinations). We also need to generate a file with the valid Go formatting for usage in pipelines, so there shouldn't be redundant imports.